components/Nav/Nav.tsx

```diff
@@ -27,6 +32,33 @@ export const Nav = (): JSX.Element => {
   const pathname = usePathname();
   const { logoutAccount } = useAuthContext();
   const [open, setOpen] = React.useState(false);
+  const { user } = useDataStore((state) => state);
+  const [leagues, setLeagues] = useState<ILeague[]>([]);
+  const [loadingLeagues, setLoadingLeagues] = useState<boolean>(true);
+
+  /**
+   * Fetches the user's leagues.
+   * @returns {Promise<void>}
+   */
+  const getLeagues = async (): Promise<void> => {
+    try {
+      const userLeagues = await getUserLeagues(user.leagues);
+      setLeagues(userLeagues);
+    } catch (error) {
+      console.error('Error fetching user leagues', error);
+    } finally {
+      setLoadingLeagues(false);
+    }
+  };
+
+  useEffect(() => {
+    if (!user.id || user.id === '') {
+      setLoadingLeagues(false);
```

Comment:
Why are we setting loadingLeagues to false here?  This would cause the sidebar to load when the user.id doesn't exist or is an empty string.   I think we can remove this here.

Reply:
Will change that shortly! Thank you.